module/project/templateTasks/helper.js

```diff
@@ -0,0 +1,638 @@
+/**
+ * name : helper.js
+ * author : Aman
+ * created-date : 22-July-2020
+ * Description : Project templates tasks helper functionality.
+ */
+
+/**
+    * ProjectTemplateTasksHelper
+    * @class
+*/
+
+// Dependencies
+const projectTemplatesHelper = require(MODULES_BASE_PATH + "/project/templates/helper");
+const kendraService = require(GENERICS_FILES_PATH + "/services/kendra");
+
+
+module.exports = class ProjectTemplateTasksHelper {
+
+    /**
+     * Lists of tasks.
+     * @method
+     * @name taskDocuments
+     * @param {Array} [filterData = "all"] - tasks filter query.
+     * @param {Array} [fieldsArray = "all"] - projected fields.
+     * @param {Array} [skipFields = "none"] - field not to include
+     * @returns {Array} Lists of tasks. 
+     */
+    
+    static taskDocuments(
+        filterData = "all", 
+        fieldsArray = "all",
+        skipFields = "none"
+    ) {
+        return new Promise(async (resolve, reject) => {
+            try {
+                
+                let queryObject = (filterData != "all") ? filterData : {};
+                let projection = {}
+           
+                if (fieldsArray != "all") {
+                    fieldsArray.forEach(field => {
+                        projection[field] = 1;
+                   });
+               }
+               
+               if( skipFields !== "none" ) {
+                   skipFields.forEach(field=>{
+                       projection[field] = 0;
+                   });
+               }
+               
+               let tasks = 
+               await database.models.projectTemplateTasks.find(
+                   queryObject, 
+                   projection
+               ).lean();
+           
+               return resolve(tasks);
+           
+           } catch (error) {
+               return reject(error);
+           }
+       });
+    }
+
+    /**
+     * Extract csv information.
+     * @method
+     * @name extractCsvInformation
+     * @param {Array} csvData - csv data.
+     * @param {String} projectTemplateId - project template id.
+     * @returns {Array} Lists of tasks.
+     */
+    
+    static extractCsvInformation(
+        csvData,
+        projectTemplateId
+    ) {
+        return new Promise(async (resolve, reject) => {
+            try {
+
+                let taskIds = [];
+                let solutionIds = [];
+                let systemId = false;
+
+                csvData.forEach(data=>{
+                    
+                    let parsedData = UTILS.valueParser(data);
+
+                    if( parsedData._SYSTEM_ID ) {
+                        taskIds.push(parsedData._SYSTEM_ID);
+                        systemId = true;
+                    } else {
+                        taskIds.push(parsedData.externalId);
+                    }
+
+                    if ( parsedData.solutionId && parsedData.solutionId !== "" ) {
+                        solutionIds.push(parsedData.solutionId);
```

Comment:
@aman-tunerlabs - Let solutionId be in URL query parameter only not in every row of CSV